src/blocker.rs

```diff
@@ -344,7 +366,11 @@ impl Blocker {
 
                 // Redirects are independent of blocking behavior.
                 if filter.is_redirect() {
-                    redirects.push(filter.clone());
+                    let mut filter = filter.clone();
+                    if filter.is_redirect_url() {
+                        filter.mask.set(crate::filters::network::NetworkFilterMask::IS_REDIRECT_URL, true);
```

Comment:
Updating the mask should occur at parse-time, not when adding to the engine.

<sub>(I know, I just finally removed a mask update here regarding `important` - there was no way around that when initially implemented... But this should work fine at parse time.)</sub>

Reply:
I think this is extraneous actually, I already update the mask in network.rs